frontend/src/components/ExpenseList.js

```diff
@@ -4,20 +4,57 @@ import { FaSave } from "react-icons/fa";
 import "../styles/ExpenseList.css";
 
 class ExpenseList extends React.Component {
+  state = {
+    userexpense: { categoryName: "", amount: "" },
+  };
+
+  onChange = (e) => {
+    // this changes the state of userexpenses amount.
+    this.setState((prevState) => {
+      let userexpense = Object.assign({}, prevState.userexpense);
+      userexpense[e.target.name] = e.target.value;
+      return { userexpense };
+    });
+  };
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    if (e.target.checkValidity()) {
+      this.props.updateTotal(this.state.userexpense.amount);
+      e.target.reset();
+    }
+  };
+
   render() {
     return (
       <>
-        <Form className="expenseform">
+        <Form noValidate onSubmit={this.onSubmit} className="expenseform">
```

Comment:
One more thing came to mind - let's allow the validate before submit (remove noValidate here and the checkValidity() call in the onSubmit function). This way there will be a default browser notification for validation errors i.e. better usability.

Reply:
Made this change, its working nicely.